### PR TITLE
Projects/add project/summary

### PR DESCRIPTION
**Pull request by amritkv**
PR Description :

1. Added Add Project's summary page UI.
2. Integrated with backend with basic data (General Info) while creating a project in "Add Project".

Note : 

1. Tried to reduce the PR size as small as possible, that's why only part of "Add Project" feature is added.
2. Due to some changes in our siemens side, we have implemented this "Add Project" api call using token. Soon it will be replaced by session token.

### CODE REVIEW

File: src/app/[locale]/projects/add/page.tsx [hunk 1]
```diff
@@ -7,456 +7,133 @@
...
+        const response = await ApiUtils.POST('projects',
+                                              projectPayload,
+                                              AUTH_TOKEN)
+
+        if (response.status == HttpStatus.CREATED) {
+            const responseData = await response.json()
+            alert(t('Your project is created'), 'success')
+            router.push('/projects')
+        } else {
+            alert(t('There are some errors while creating project'), 'error')
```
Comment (hoangnt2) [inline src/app/[locale]/projects/add/page.tsx:61]: Missing key **There are some errors while creating project** in language files

File: src/app/[locale]/projects/add/components/Summary/Summary.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+interface Props{
+    token: string
+    projectPayload: ProjectPayload
+    setProjectPayload: React.Dispatch<React.SetStateAction<ProjectPayload>>
+}
+
+export default function Summary({token, projectPayload, setProjectPayload}: Props) {
+
+    const t = useTranslations(COMMON_NAMESPACE)
+    const [additionalDataList, setAdditionalDataList] = useState<AdditionalDataInput[]>([
```
Comment (hoangnt2) [inline src/app/[locale]/projects/add/components/Summary/Summary.tsx:33]: default additionalDataList should be empty [ ]

File: src/app/[locale]/projects/add/page.tsx [hunk 3]
```diff
@@ -7,456 +7,133 @@
...
+    const t = useTranslations(COMMON_NAMESPACE)
+    const [projectPayload, setProjectPayload] = useState<ProjectPayload>({
+        name: '',
+        description: '',
+        version: '',
+        visibility: '',
+        projectType: '',
+        tag: '',
+        domain: '',
+        leadArchitect: 'test@mail.com',
```
Comment (hoangnt2) [inline src/app/[locale]/projects/add/page.tsx:37]: remove test email

**Comment by hoangnt2** [comment]
@amritkv, I don't see this one in sw360 add project page 
image

By the way, you should move the **AddSummary** component to /src/components folder, it can be reused by edit page

**Comment by amritkv** [comment]
> setAdditionalDataList
@hoangnt2 
In our side we have this `Add Additional Data` field, 
Ohh may be the case that there is no default value.
Can you please share once, what comes in this section at your end ?

**Comment by hoangnt2** [comment]
@amritkv, I don't see them, maybe this is an option

image

**Comment by amritkv** [comment]
> @amritkv, I don't see them, maybe this is an option
> 
> image

Ohh now I got it. 👍

File: src/app/[locale]/projects/add/page.tsx (hunk 1 above)
Comment (amritkv) [reply src/app/[locale]/projects/add/page.tsx:61]: Added in the language file.

File: src/app/[locale]/projects/add/components/Summary/Summary.tsx (hunk 2 above)
Comment (amritkv) [reply src/app/[locale]/projects/add/components/Summary/Summary.tsx]: Removed and used generic component

File: src/app/[locale]/projects/add/page.tsx (hunk 3 above)
Comment (amritkv) [reply src/app/[locale]/projects/add/page.tsx]: Done

**Comment by hoangnt2** [comment]
When I tested the GUI, I had this problem. The styling doesn't seem to work as expected. 
Do you have a problem like this on your side?

Screenshot from 2023-08-09 18-00-11

**Comment by amritkv** [comment]
> @amritkv When I test the GUI I have this problem. Styling doesn't seem to work like the old GUI (alignment). Do you have the problem like this on your side?
> 
> Screenshot from 2023-08-09 18-00-11

@hoangnt2 Yeah it's the same as it's on your side. We will fix this on the go along with adding the features to the project page.

**Comment by hoangnt2** [comment]
> > @amritkv When I test the GUI I have this problem. Styling doesn't seem to work like the old GUI (alignment). Do you have the problem like this on your side?
> > Screenshot from 2023-08-09 18-00-11
> 
> @hoangnt2 Yeah it's the same as it's on your side. We will fix this on the go along with adding the features to the project page.

OK please fix GUI styling in next PR, and I think you can move **Summary** component to **/src/components** folder. It can be used by edit component to

**Comment by amritkv** [comment]
> > > @amritkv When I test the GUI I have this problem. Styling doesn't seem to work like the old GUI (alignment). Do you have the problem like this on your side?
> > > Screenshot from 2023-08-09 18-00-11
> > 
> > 
> > @hoangnt2 Yeah it's the same as it's on your side. We will fix this on the go along with adding the features to the project page.
> 
> OK please fix GUI styling in next PR, and I think you can move **Summary** component to **/src/components** folder. It can be used by edit project page too

@hoangnt2 Yupp sure ! Will do that.
And yeah its a great idea to move it to **/src/components**

**Comment by amritkv** [comment]
> > > @amritkv When I test the GUI I have this problem. Styling doesn't seem to work like the old GUI (alignment). Do you have the problem like this on your side?
> > > Screenshot from 2023-08-09 18-00-11
> > 
> > 
> > @hoangnt2 Yeah it's the same as it's on your side. We will fix this on the go along with adding the features to the project page.
> 
> OK please fix GUI styling in next PR, and I think you can move **Summary** component to **/src/components** folder. It can be used by edit project page too

@hoangnt2 Fixed this UI alignment issues in this PR itself.

Here is the new look : 

image

**Review by heliocastro — APPROVED** [review]
Good to go
Next commit -> Small ones

**Comment by heliocastro** [comment]
> Good to go Next commit -> Small ones

Nope, build still breaking.
@amritkv Check the log output

**Comment by amritkv** [comment]
> > Good to go Next commit -> Small ones
> 
> Nope, build still breaking. @amritkv Check the log output

Ready to get merged.

**Comment by hoangnt2** [comment]
@amritkv Add project page seem not work, It look like you are missing some props when you use **AddKeyValue** component
You can refer to **ComponentAddSummary** component to fix it or simply replace error line to
`props.data?.map((elem, j) => {`

image

Do the same thing with  **AddAdditionalRoles** component

Currently It seem look like you can't add these properties to project. I hope you will do this in next PR

**Comment by hoangnt2** [comment]
@amritkv, and please revert button's css, the border are disappeared
image

**Comment by amritkv** [comment]
> @amritkv Add project page seem not work, It look like you are missing some props when you use **AddKeyValue** component You can refer to **ComponentAddSummary** component to fix it or simply replace error line to `props.data?.map((elem, j) => {`
> 
> image
> 
> Do the same thing with **AddAdditionalRoles** component
> 
> Currently It seem look like you can't add these properties to project. I hope you will do this in next PR

Ohhkk I got it. Here in these two files `AddAdditionalRoles` and `AddKeyValue` we need to modify a small code section. I will use different version of these two files separately for now to keep the component page working. Then in next we can discuss and make single  `AddAdditionalRoles` and `AddKeyValue` files for both component and project page.

File: src/app/[locale]/projects/add/page.tsx [hunk 4]
```diff
@@ -7,456 +7,133 @@
...
+
+export default function AddProjects() {
+
+    const router = useRouter();
+    const t = useTranslations(COMMON_NAMESPACE)
+    const [projectPayload, setProjectPayload] = useState<ProjectPayload>({
+        name: '',
+        description: '',
+        version: '',
+        visibility: '',
```
Comment (hoangnt2) [inline src/app/[locale]/projects/add/page.tsx:33]: **visibility** and **projectType** should have default value (default option on GUI).

**Comment by hoangnt2** [comment]
@amritkv, Everything is fine but there are some small things you need to fix then we can merge this PR
- You should set the default values for **visibility** and **projectType**. If you don't change them on the GUI these fields will not be updated (will be an empty string) and the project can't be created.

- There are some missing keys in language files
image

- And please add the **\*** on the required fields
Screenshot from 2023-08-17 14-11-15

File: src/app/[locale]/projects/add/page.tsx (hunk 4 above)
Comment (amritkv) [reply src/app/[locale]/projects/add/page.tsx]: Resolved ! 👍

**Comment by amritkv** [comment]
> @amritkv, Everything is fine but there are some small things you need to fix then we can merge this PR
> 
> * You should set the default values for **visibility** and **projectType**. If you don't change them on the GUI these fields will not be updated (will be an empty string) and the project can't be created.
> * There are some missing keys in language files
>   image
> * And please add the ***** on the required fields
>   Screenshot from 2023-08-17 14-11-15

Resolved ! 👍

**Comment by hoangnt2** [comment]
Look good

**Review by hoangnt2 — APPROVED** [review]

**Pull request merged**